extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -615,31 +604,47 @@ export class AmpA4A extends AMP.BaseElement {
           //    render-in-DOM failed, and no ad would be displayed.  However,
           //    all of the enclosed mutations are fairly simple and unlikely
           //    to fail.
-          this.vsync_.mutate(() => {
-            const doc = this.element.ownerDocument;
-            // Copy fonts to host document head.
-            this.relocateFonts_(creativeMetaData);
-            // Create and setup shadow root.
-            const shadowRoot = createShadowEmbedRoot(this.element,
-                creativeMetaData.customElementExtensions || []);
-            // Add custom CSS.
-            const customStyle = doc.createElement('style');
-            customStyle.setAttribute('amp-custom', '');
-            customStyle.textContent = cssBlock;
-            shadowRoot.appendChild(customStyle);
-            // Add body.
-            const bodyAttrString = creativeMetaData.bodyAttributes ?
-                  ' ' + creativeMetaData.bodyAttributes : '';
-            const temp = doc.createElement('div');
-            temp./*OK*/innerHTML =
-                `<${AMP_BODY_STRING}${bodyAttrString}></${AMP_BODY_STRING}>`;
-            const bodyElement = temp.firstElementChild;
-            shadowRoot.appendChild(bodyElement);
-            bodyElement./*OK*/innerHTML = bodyBlock;
-            this.rendered_ = true;
-            this.onAmpCreativeShadowDomRender();
+          return this.vsync_.mutatePromise().then(() => {
+            // Create and setup friendly iframe.
+            const iframe = this.element.ownerDocument.createElement('iframe');
+            this.applyFillContent(iframe);
+
+            iframe.setAttribute('frameborder', '0');
+            iframe.setAttribute('allowfullscreen', '');
+            iframe.setAttribute('allowtransparency', '');
+            iframe.setAttribute('scrolling', 'no');
+
+            // HTML is all but the extensions which must be removed from the
+            // creative.
+            // TODO(keithwrightbos): remove this monstrousity in favor of
+            // validation providing offsets to extension & AMP runtime
+            // locations.
+            const extensions =
+              creativeMetaData.customElementExtensions || [];
+            for (let i = 0; i <= extensions.length; i++) {
+              // Remove AMP runtime as well.
+              const extensionName =
```

Comment:
I don't quiet understand these changes. Why not just use the old way of doing this as was before, but simply combine `cssBlock` and `bodyBlock` into one string before sending it to `installFriendlyIframeEmbed`?